internal/test/fuzz_test.go

```diff
@@ -0,0 +1,222 @@
+package test
+
+import (
+	"bytes"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/iov-one/cosmos-sdk-crud/internal/store/types"
+
+	"github.com/tendermint/tendermint/libs/rand"
+
+	crud "github.com/iov-one/cosmos-sdk-crud"
+)
+
+const domainMinLength = 4
```

Comment:
Please use the starname constraint parameters from below and add `uri` and `resource` tests.
```
      "valid_domain_name": "^[-a-z0-9_]{4,32}$",
      "valid_account_name": "^[-.a-z0-9_]{1,63}$",
      "valid_uri": "^[-a-z0-9]{1,15}:[-a-z0-9]{1,40}$",
      "valid_resource": "^[-.a-z0-9A-Z/:_ #]{1,200}$",
```
(Those parameters are from our mainnet; `curl -X POST "https://lcd-private-iov-mainnet-2.iov.one/configuration/query/configuration"`)

Reply:
Do you want me to use those parameters as they are defined now or to fetch them from the mainnet in the test ?
I don't know how I have to test uri and resource, as these attributes does not seem to be secondary keys of the Starname object (if so they are not used with `cosmos-sdk-crud`)

Reply:
Just use the hard-coded parameters for now.

You're right about the `uri`, it's not indexed; however, `resource` is on the mainnet.  The Starname object isn't a perfect representation of the data on the mainnet.

Reply:
So I add a `resource` field to the `TestStarname` object in order to test it ?

Reply:
Yes, that's probably the path of least resistance.